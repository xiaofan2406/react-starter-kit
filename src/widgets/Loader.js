/**
 * @see https://github.com/yuanyan/halogen/blob/master/src/RingLoader.js
 */
import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { css } from '@emotion/core';

const Loader = ({ size, color }) => {
  const cssSize = useMemo(
    () =>
      css`
        width: ${size}px;
        height: ${size}px;
      `,
    [size]
  );

  return (
    <div
      css={css`
        display: inline-block;
      `}
    >
      <div
        css={css`
          ${cssSize};
          position: relative;

          & div {
            ${cssSize};
            border-style: solid;
            border-width: ${size / 10}px;
            border-color: ${color};
            opacity: 0.4;
            border-radius: 100%;
            position: absolute;
            top: 0;
            left: 0;
            perspective: 800px;
            animation-fill-mode: forwards;
          }
        `}
      >
        <div
          css={css`
            animation: loaderRigthRotate 2s 0s infinite linear;
          `}
        />
        <div
          css={css`
            animation: loaderLeftRotate 2s 0s infinite linear;
          `}
        />
      </div>
    </div>
  );
};

Loader.propTypes = {
  color: PropTypes.string,
  size: PropTypes.number,
};

Loader.defaultProps = {
  color: '#00bcd4',
  size: 60,
};

export default Loader;
