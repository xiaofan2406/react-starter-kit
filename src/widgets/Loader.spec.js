import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Loader from './Loader';

it('should match snapshot', () => {
  const props = { classes: {} };
  const component = mount(<Loader {...props} />);

  expect(toJson(component)).toMatchSnapshot();
});
