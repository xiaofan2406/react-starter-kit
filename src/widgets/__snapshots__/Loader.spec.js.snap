// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Loader
  color="#e8e8e8"
  size={60}
>
  <styled(ComponentLoaderContainer)
    color="#e8e8e8"
    size={60}
  >
    <div
      className="css-LoaderContainer-17fo30n vars-1hjfu8r"
      color="#e8e8e8"
      size={60}
    >
      <div>
        <div />
        <div />
      </div>
    </div>
  </styled(ComponentLoaderContainer)>
</Loader>
`;
